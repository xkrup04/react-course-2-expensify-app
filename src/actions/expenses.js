import uuid from "uuid";

// Add_expense
export const addExpense = ({
  description = "",
  note = "",
  amount = 0,
  createdAt = 0
} = {}) => ({
  type: "ADD_EXPENSE",
  expense: {
    id: uuid(),
    description,
    note,
    amount,
    createdAt
  }
});

// Remove_expense
export const removeExpense = ({ id } = {}) => ({
  type: "REMOVE_EXPENSE",
  id
});

// Edit_expense
export const editExpense = (id, updates) => ({
  type: "EDIT_EXPENSE",
  id,
  updates
});
