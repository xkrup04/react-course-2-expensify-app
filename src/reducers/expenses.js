// Expenses reducer
const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
  switch (action.type) {
    case "ADD_EXPENSE":
      // concat will create new array adding items to the end
      // return state.concat(action.expense);
      // Spread operator allows to add items before or after...
      return [...state, action.expense];
    case "REMOVE_EXPENSE":
      // filter will return new array for specified conditions
      return state.filter(({ id }) => action.id !== id);
    case "EDIT_EXPENSE":
      return state.map(expense => {
        if (expense.id === action.id) {
          return {
            //Babel spread object operator with override properties set later
            ...expense,
            ...action.updates
          };
        } else {
          return expense;
        }
      });
    default:
      return state;
  }
};

