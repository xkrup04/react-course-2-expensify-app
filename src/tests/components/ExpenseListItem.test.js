import React from "react";
import { shallow } from "enzyme";
import ExpenseListItem from "../../components/ExpenseListItem";
import expenses from "../fixtures/expenses";

test("shoud render ExpenseListItem with expense", () => {
  const expense = expenses[1];
  const wrapper = shallow(<ExpenseListItem {...expense} />);
  expect(wrapper).toMatchSnapshot();
});
