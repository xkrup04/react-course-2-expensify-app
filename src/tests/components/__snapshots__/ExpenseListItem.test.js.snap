// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoud render ExpenseListItem with expense 1`] = `
<div>
  <Link
    to="/edit/2"
  >
    <h1>
      Rent
    </h1>
  </Link>
  <p>
    109500
     
    -345600000
  </p>
</div>
`;
